### PR TITLE
Added inmemory caching to all samples

### PR DESCRIPTION
**Pull request by v-michaelmi**
## Purpose

* Updated applications to use **AddInMemoryTokenCache** to prevent redundant calls for retrieving access tokens

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

The samples still work the same but now utilize a memory cache for tokens.

See reference documentation [here](https://docs.microsoft.com/en-us/azure/active-directory/develop/msal-net-token-cache-serialization?tabs=aspnetcore).

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[x] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code
*  Run all samples and ensure they worked the same as before

* Test the code

```
Follow the instructions in each sample.
```

## What to Check
Verify that the following are valid
* Each sample works the same as before
* Samples utilize tokens in the cache if another call is made to retrieve a token.

```
            // Acquire a token (twice)
            var result = await app.AcquireTokenForClient(scopes)
                .ExecuteAsync();
            Console.WriteLine(result.AuthenticationResultMetadata.TokenSource);

            result = await app.AcquireTokenForClient(scopes)
                .ExecuteAsync();
                
           /*
            * If the token has not been acquired from sever this should output:
            * 
            * > IdentityProvider
            * > Cache
            */
```

### CODE REVIEW

File: 1-Call-MSGraph/daemon-console/daemon-console.csproj [hunk 1]
```diff
@@ -13,7 +13,8 @@
   <ItemGroup>
     <PackageReference Include="Microsoft.Extensions.Configuration.Binder" Version="5.0.0" />
     <PackageReference Include="Microsoft.Extensions.Configuration.Json" Version="3.1.0" />
-    <PackageReference Include="Microsoft.Identity.Web.Certificate" Version="1.18.0" />
+    <PackageReference Include="Microsoft.Identity.Web" Version="1.21.0" />
```
Comment (jmprieur) [inline 1-Call-MSGraph/daemon-console/daemon-console.csproj:16]: Why is Microsoft.Identity.Web needed?

File: 2-Call-OwnApi/daemon-console/Daemon-Console.csproj [hunk 2]
```diff
@@ -12,7 +12,8 @@
 
 	<ItemGroup>
 		<PackageReference Include="Microsoft.Extensions.Configuration.Binder" Version="5.0.0" />
-		<PackageReference Include="Microsoft.Identity.Web.Certificate" Version="1.18.0" />
+		<PackageReference Include="Microsoft.Identity.Web" Version="1.21.0" />
```
Comment (jmprieur) [inline 2-Call-OwnApi/daemon-console/Daemon-Console.csproj:15]: same question

File: 3-Using-KeyVault/daemon-console/daemon-console.csproj [hunk 3]
```diff
@@ -14,7 +14,8 @@
     <PackageReference Include="Microsoft.Extensions.Configuration" Version="5.0.0" />
     <PackageReference Include="Microsoft.Extensions.Configuration.Binder" Version="5.0.0" />
     <PackageReference Include="Microsoft.Extensions.Configuration.Json" Version="5.0.0" />
-    <PackageReference Include="Microsoft.Identity.Web.Certificate" Version="1.18.0" />
+    <PackageReference Include="Microsoft.Identity.Web" Version="1.21.0" />
```
Comment (jmprieur) [inline 3-Using-KeyVault/daemon-console/daemon-console.csproj:17]: Same

File: 4-Call-OwnApi-Pop/daemon-console/Daemon-Console.csproj [hunk 4]
```diff
@@ -14,7 +14,8 @@
     <PackageReference Include="Microsoft.Extensions.Configuration" Version="3.0.0" />
     <PackageReference Include="Microsoft.Extensions.Configuration.Binder" Version="3.0.0" />
     <PackageReference Include="Microsoft.Extensions.Configuration.Json" Version="3.0.0" />
-    <PackageReference Include="Microsoft.Identity.Client" Version="4.28.1" />
+    <PackageReference Include="Microsoft.Identity.Client" Version="4.38.0" />
+    <PackageReference Include="Microsoft.Identity.Web" Version="1.21.0" />
```
Comment (jmprieur) [inline 4-Call-OwnApi-Pop/daemon-console/Daemon-Console.csproj:18]: Same?

**Review by jmprieur — COMMENTED** [review]
@v-michaelmi : do we need Microsoft.Identity.Web ? Microsoft.Identity.Web.TokenCache should be enough?

**Comment by v-michaelmi** [comment]
@jmprieur 

That was indeed overkill. Updated with latest commit.

Thanks for pointing it out!

**Review by jmprieur — APPROVED** [review]
LGTM
Thanks @v-michaelmi

**Pull request merged**